internal/service/ecs/task_definition.go

```diff
@@ -768,6 +814,18 @@ func expandEcsTaskDefinitionPlacementConstraints(constraints []interface{}) ([]*
 	return pc, nil
 }
 
+func expandEcsTaskDefinitionRuntimePlatformConfiguration(runtimePlatformConfig []interface{}) *ecs.RuntimePlatform {
+	config := runtimePlatformConfig[0]
+	configMap := config.(map[string]interface{})
+
+	ecsProxyConfig := &ecs.RuntimePlatform{
```

Comment:
taking above into account only one of these (or even none) can provided. so lets only set each of the arguments if there are provided. this guard against nil defernce

Reply:
Looking at `expandEcsTaskDefinitionRuntimePlatformConfiguration` the typelists is passed in as an interface and then as its an array of interface type we take the first entry... fine. But if you set neither property then the resulting entry is nil; but the cast to map[string]interface happens there after which would be an exception. 

I've added a nill check to return empty runtime_platform if the config variable is nil. But I'm wondering if I should as the rest of the similar objects dont. If you create a runtime_platform object with no properties in it, terraform plan shows the change each time

```
          # aws_ecs_task_definition.test must be replaced
              ~ revision                 = 1 -> (known after apply)
              ~ tags_all                 = {} -> (known after apply)
                # (5 unchanged attributes hidden)

              + runtime_platform { # forces replacement}
            }

        Plan: 1 to add, 0 to change, 1 to destroy.
```

I may be overthinking this as technically its not a valid config; but the config validation across ECS is already a bit vague.

Reply:
Ignore that, thinking aloud.